datasets/preprocess_utils.py

```diff
@@ -243,3 +243,181 @@ def bin_stream_hilmi24(
     feat_count = np.concatenate([feat_count_ab, feat_count_bl])
 
     return phi1_bcent, feat_mean, feat_stdv, feat_count
+
+
+# Get simulated Gaia G band magnitudes
+def gaia_transform(g, r, i, z, version='dr3'):
+    """From Eli Rykoff for FGCM comparisons via Slack:
+    https://darkenergysurvey.slack.com/archives/C016J5SDV9T/p1608182260343200
+
+    This complex model behaves about as well as a random forest
+    classifier for Gaia DR2.  The magnitude dependence of the
+    transformation is huge because of background errors in Gaia DR2.
+
+    The EDR3 transformations have a much smaller r-offset curvature
+    due to background issues and should be able to go deeper.
+
+    These transformations are valid for 0 < g - i < 1.5
+
+    /nfs/slac/kipac/fs1/g/des/erykoff/des/y6a1/fgcm/run_v2.1.2/gedr3/gaia_superfitmodel3.py
+
+    Parameters
+    ----------
+    g, r, i, z : DECam magnitudes in g,r,i,z  (AB system)
+    version    : Version of the Gaia catalog to compare against
+
+    Returns
+    -------
+    Gmag  : Predicted Gaia G-band magnitude based on DECam griz
+    """
+
+    magConst = 2.5 / np.log(10.0)
+    lambdaStd = np.array([4790.28076172, 6403.26367188, 7802.49755859, 9158.77441406])
+    i0Std = np.array([0.16008162, 0.18297842, 0.17169334, 0.1337308])
+    i1Std = np.array([2.09808350e-05, -1.22070312e-04, -1.08942389e-04, -8.01086426e-05])
+    i10Std = i1Std / i0Std
+    fudgeFactors = np.array([0.25, 1.0, 1.0, 0.25])
+    fudgeShift   = 0.0 # Additive shift in peak (mag)
+    nBands = lambdaStd.size
+
+    #g -= -46 * 1e-3
+    #r -= 0   * 1e-3
+    #i -= 38  * 1e-3
+    #z -= 31  * 1e-3
+
+    trainCat = np.rec.fromarrays([g,r,i,z],names=['g','r','i','z'])
+    fluxg = 10**(g/-2.5)
+    fluxr = 10**(r/-2.5)
+    fluxi = 10**(i/-2.5)
+    fluxz = 10**(z/-2.5)
+
+    S = np.zeros((trainCat.size, nBands - 1), dtype='f8')
+
+    S[:, 0] = (-1. / magConst) * (trainCat['r'] - trainCat['g']) / (lambdaStd[1] - lambdaStd[0])
+    S[:, 1] = (-1. / magConst) * (trainCat['i'] - trainCat['r']) / (lambdaStd[2] - lambdaStd[1])
+    S[:, 2] = (-1. / magConst) * (trainCat['z'] - trainCat['i']) / (lambdaStd[3] - lambdaStd[2])
+
+    fnuPrime = np.zeros((trainCat.size, nBands))
+    fnuPrime[:, 0] = S[:, 0] + fudgeFactors[0] * (S[:, 1] + S[:, 0])
+    fnuPrime[:, 1] = fudgeFactors[1] * (S[:, 0] + S[:, 1]) / 2.0
+    fnuPrime[:, 2] = fudgeFactors[2] * (S[:, 1] + S[:, 2]) / 2.0
+    fnuPrime[:, 3] = S[:, 2] + fudgeFactors[3] * ((lambdaStd[3] - lambdaStd[2]) / (lambdaStd[3] - lambdaStd[1])) * (S[:, 2] - S[:, 1])
```

Comment:
Try to limit the number of characters per line to 80-100 max. I'm not always strict with it, but this line is too long.

Reply:
fixed